File: LeetCode/921.MinimumAdd_to_Make_Parentheses_Valid/susan/921.MinimumAdd_to_Make_Parentheses_Valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int right = 0;
        int left = 0;
        for(char c : s.toCharArray()){
            if(c == '('){
                right++;
            }
            if(c == ')'){
                if(right == 0){
                    left++;
                }else{
                    right--;
                }
            }
        }
        return right+left;
    }
}

File: LeetCode/921.Minimum_Add_to_Make_Parentheses_Valid/921.Minimum_Add_to_Make_Parentheses_Valid.py
#!/usr/bin/env python3


class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        # valid:
        # "" / AB / (A)
        # one move: insert parenthesis

        left, right = 0, 0
        res = 0
        for i, c in enumerate(s):
            if c == "(":
                left += 1
            elif c == ")":
                if left:
                    left -= 1
                else:
                    right += 1

        return left + right
